spec/async_producer_spec.rb

```diff
@@ -30,6 +31,7 @@ def instrument(name, payload = {})
       max_retries: 2,
       retry_backoff: 0.2,
       logger: logger,
+      finally: lambda { | messages | lambda_state["has_been_called"] = true }
```

Comment:
@ferrous26 do you think this pattern can work? 

The idea is to define `lambda_state` which is mutable, include it in the lambda's closure, then pass the lambda into the producer. When it's called it should modify the `lambda_state` which should be observable in the scope of the spec test.

The test as is is failing with expected true but got false so either the lambda isn't being called or the scope isn't working how I expected. Any advice?